cpp/unreal_plugins/UrdfBot/Source/UrdfParser.h

```diff
@@ -0,0 +1,142 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#pragma once
+
+#include <Eigen/Dense>
+#include <map>
+#include <vector>
+
+#include <CoreMinimal.h>
+
+class FXmlNode;
+
+struct UrdfJointDesc;
+struct UrdfLinkDesc;
+struct UrdfMaterialDesc;
+
+enum UrdfGeometryType
+{
+    INVALID_GEOMETRY = 0,
+    BOX,
+    SPHERE,
+    CYLINDER,
+    MESH
+};
+
+enum UrdfJointType
+{
+    // Some of these (such as FIXED) are already macros
+    INVALID_TYPE = 0,
+    REVOLUTE_TYPE,
+    CONTINUOUS_TYPE,
+    PRISMATIC_TYPE,
+    FIXED_TYPE,
+    FLOATING_TYPE,
+    PLANAR_TYPE
+};
+
+struct UrdfGeometry
```

Comment:
- `UrdfGeometryDesc` for consistency with other structs
- I know I've gone back and forth on this, but let's define one additional level of hierarchy here, with a `UrdfMeshDesc` struct which has a `filename_` and a `scale_` parameter.
- We should define sensible default values for all of the primitive types in these structs.

Reply:
fixed. The default shape for sphere and cylinder use LoCoBot Wheel value. Box uses the same size for consistancy.